### ISSUE TITLE
argmax between generator and discriminator

### ISSUE THREAD

**Issue by stefat77** [eriklindernoren/Keras-GAN#61]
Hello,
I have a silly question... I'm trying to make a GAN like this:

input(h,w,c) --> [GEN] --> out1(h,w,2) --> [LambdaLayer] --> out2(h,w,1) --> [DISCRIMINATOR] --> val

The problem is that the Lambda function is the following:
out2 = Lambda(lambda x: K.cast(K.expand_dims(K.argmax(x)), dtype="float32"))(out1)
It calls the argmax function that is not differentiable so the backprop is not working correctly.

out1 is the output mask from the generator/segmentation net, and every pixel describe the probability distribution over 2 classes.
I need the argmax in between in order to compare the out1 mask with some other external masks through the discriminator network.
How can I solve the problem?
